src/cli/util.js

```diff
@@ -441,6 +443,18 @@ function formatFiles(context) {
     context.logger.log("Checking formatting...");
   }
 
+  let changedCache = null;
+  if (context.argv["only-changed"]) {
+    const cacheDir = findCacheDir({ name: "prettier", create: true });
```

Comment:
/cc @g-harel need use `|| os.tmp()` https://github.com/avajs/find-cache-dir#findcachediroptions

> It returns a string containing the absolute path to the cache directory, or undefined if package.json was never found.

In theory you can install `prettier` globally and doesn't have `package.json`

Reply:
Nice catch! I will open up a PR with a fix.

I'm thinking it might be easier to leave `changedCache` null if `findCacheDir` fails.